src/main/java/Chatbot/alain/Storage.java

```diff
@@ -0,0 +1,212 @@
+            String[] dateAndTime = inputTime.split(" ");
+            String addMsg = "";
+            for (int i = 1; i < dateAndTime.length; i++) {
+                addMsg += dateAndTime[i];
+            }
+            DateTimeFormatter inputPattern = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+            LocalDate date = LocalDate.parse(dateAndTime[0], inputPattern);
+
+            DateTimeFormatter outputPattern = DateTimeFormatter.ofPattern("MMMM dd yyyy", Locale.ENGLISH);
+            String transformedTime = date.format(outputPattern);
+            return transformedTime.toString() + " " + addMsg;
+        } else if (inputTime.length() == 0) {
+            throw new AlainException(" OOPS!!! The description of a alain.Task cannot be empty.");
+        } else {
+            return inputTime;
+        }
+    }
+
+    /**
+     * Loads tasks from a file and processes user input.
+     *
+     * @return The task list loaded from the file.
+     * @throws IOException If an I/O error occurs.
+     */
+    public TaskList loadTasksFromFile() throws IOException, AlainException {
+        ui.showWelcome();
+        ArrayList<Task> tasks = new ArrayList<>();
+        TaskList list = new TaskList(tasks);
+        try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            String text;
+            while ((text = reader.readLine()) != null) {
+                boolean isMatchMark = Pattern.matches("mark \\d+", text);
+                boolean isMatchUnmark = Pattern.matches("unmark \\d+", text);
+                boolean isDeadline = Pattern.matches("deadline .+", text);
+                boolean isToDo = Pattern.matches("todo .+", text);
+                boolean isEvent = Pattern.matches("event .+", text);
+                boolean isDelete = Pattern.matches("delete .+", text);
+                boolean isFind = Pattern.matches("find .+", text);
+                if (isFind) {
+                    String keyWord = text.substring(4);
+                    TaskList tmpList = new TaskList();
+                    for (int i = 0; i < list.size(); i++) {
+                        if (list.getTask(i).descriptionContain(keyWord)) {
+                            tmpList.addTask(list.getTask(i));
+                        }
+                    }
+                    ui.showListContainingKeyword(tmpList);
+                    continue;
+                }
+                if (isDelete) {
+                    String numericPart = text.substring(7);
+                    int pos = Integer.parseInt(numericPart) - 1;
+                    if (pos >= 0 && pos < list.size()) {
+                        Task removedTask = list.removeTask(pos);
+                        ui.showRemoveTask(removedTask, list);
+                    } else {
+                        throw new AlainException("Invalid task index.");
+                    }
+                    continue;
+                }
+                if (isToDo) {
```

Comment:
Perhaps the use of if, else if, else (or) switch-case structure would have been more suitable?